folium/features.py

```diff
@@ -414,8 +414,10 @@ def __init__(self, data, style_function=None, name=None,
             if hasattr(data, 'to_crs'):
                 data = data.to_crs(epsg='4326')
             self.data = json.loads(json.dumps(data.__geo_interface__))  # noqa
+        elif hasattr(data, 'geometry') and any(data['geometry'].isnull()):  # for GeoDataFrames with missing geometries
```

Comment:
This needs to be nested inside the `elif` above or it won't be executed.
You can also drop the comment.

Reply:
But GeoDataFrames with missing geometries have no `__geometry_interface__` as I noted in #1037, so it should be executed, no?

Reply:
Indeed. I forgot about that. Sorry, my mind is one the generic objects that implement that and should fail if not. That is why the dual message is unnecessary IMO. Anyway, if you are happy with this we can merge it.

Reply:
Okay, I think it's ready.